assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js

```diff
@@ -292,11 +292,17 @@ export default function UserDimensionsPieChart( { dimensionName, dimensionValue,
 		: { __html: '' };
 
 	const options = { ...UserDimensionsPieChart.chartOptions };
+
 	if ( report?.[ 0 ]?.data?.rows?.length < 2 ) {
 		// Hide pie slice text when there is just one slice because it will overlap with the chart title.
 		options.pieSliceTextStyle.color = 'transparent';
 	}
 
+	if ( report?.[0]?.data?.rows?.length === 1 ) {
+		// When there is only one row, the chart will add a label which we need to hide - see issue #2660
+		options.pieSliceText = 'none';
+	}
```

Comment:
@johnPhillips does this condition need to be separate than the one above? They both seem to cater for the single slice case and do different things. Is there any reason to change the `pieSliceTextStyle.color` in this case if we're changing the `pieSliceText` to `none` ?

Reply:
I spoke to @eugene-manuilov who wrote the other condition here and it seems they are both attacking the same problem. I've combined them.